src/main/scala/is/hail/expr/types/TArray.scala

```diff
@@ -44,7 +44,7 @@ final case class TArray(elementType: Type, override val required: Boolean = fals
   def _typeCheck(a: Any): Boolean = a.isInstanceOf[IndexedSeq[_]] &&
     a.asInstanceOf[IndexedSeq[_]].forall(elementType.typeCheck)
 
-  override def str(a: Annotation): String = JsonMethods.compact(toJSON(a))
+  override def strNonNull(a: Annotation, config: PrintConfig): String = JsonMethods.compact(toJSON(a))
```

Comment:
Unrelated to the PrintConfig discussion, I think we do need to manually construct this string because numbers inside arrays should be printed the same as numbers not inside arrays, right? `JsonMethods.compact` will format numbers in whatever way it sees fit.

Reply:
I totally agree, and voiced this in an early discussion with Cotton + Jackie about TTuples. Cotton thinks that we need to be able to parse our own str'ed values, so we'd need to write a parser at the same time to do that. I'm not sure I agree that we need to be able to parse our str output.

Reply:
not going to touch that here, though

Reply:
💯 `lambda x: read(write(x))` should be the identity!